### PR TITLE
Reduce the size of websocket updates for transactions.

### PR DESCRIPTION
**Pull request by matthew1001**
Subscribers can use /transactions/{txid} to provide full detail about a transaction if required.

Delivered as part of closing https://github.com/hyperledger/firefly/issues/1108

Signed-off-by: Matthew Whitehead <matthew.whitehead@kaleido.io>

### CODE REVIEW

File: pkg/ffcapi/api.go
Comment (peterbroadhurst) [inline pkg/ffcapi/api.go:137]: I think this is a duplicate from https://github.com/hyperledger/firefly-transaction-manager/pull/43 - so you'll need an update once that merges.

File: pkg/fftm/policyloop.go
Comment (peterbroadhurst) [inline pkg/fftm/policyloop.go:288]: Think this function needs a name change, as it's a conditional processing.
Maybe `checkSendWSReply` or `sendWSReplyIfResolved`

File: pkg/ffcapi/api.go
Comment (matthew1001) [reply pkg/ffcapi/api.go:137]: Yeah I used the same function from #43 knowing we would be merging around the same time. Happy to do the merge once #43 is in.

File: pkg/fftm/policyloop.go
Comment (matthew1001) [reply pkg/fftm/policyloop.go:292]: Yes agreed. Because I'm not a fan of `do-a-thing-if-a-condition-is-met` function names I've chosen to put the status check outside the `sendWSReply` function call.

File: pkg/fftm/policyloop.go
Comment (peterbroadhurst) [inline pkg/fftm/policyloop.go:344]: This feels a little unnecessary to me. It extends the difference between notifications and normal just `GET` `/transactions` output, which are really different for nothing but historical reasons (EthConnect).

However, happy to leave that with you for a final decision @matthew1001

**Review by peterbroadhurst — APPROVED** [review]
One comment for you to consider, with approval

File: pkg/fftm/policyloop.go
Comment (peterbroadhurst) [reply pkg/fftm/policyloop.go:344]: (specifically I mean the pulling one field out of `receipt` up to the top-level, vs. the app being notified accessing it at `receipt.protocolId`)

Comment (matthew1001) [reply pkg/fftm/policyloop.go:344]: I think it makes sense to me, given it's expect in the notification. Suggest leaving for now and can go back round the loop in a future PR?

**Pull request merged**
